supabase/migrations/06_connectors.sql

```diff
@@ -30,11 +33,17 @@ comment on column connectors.open_graph_raw is
   'Open-graph metadata as returned by the external_url';
 comment on column connectors.open_graph_patch is
   'Patches to open-graph metadata, as a JSON merge patch';
+comment on column connectors.oauth2_client_id is
+  'oauth client id';
+comment on column connectors.oauth2_client_secret is
+  'oauth client secret';
+comment on column connectors.oauth2_spec is
+  'OAuth2 specification of the connector';
 
--- authenticated may select all connectors.
-grant select  on table connectors to authenticated;
--- But don't expose details of open_graph raw responses & patching.
-revoke select (open_graph_raw, open_graph_patch) on table connectors from authenticated;
+-- don't expose details of open_graph raw responses & patching and oauth2 secret
+revoke select on table connectors from authenticated;
+-- authenticated may select other columns for all connectors connectors.
+grant select(id, detail, updated_at, created_at, image_name, external_url, open_graph, oauth2_client_id) on table connectors to authenticated;
```

Comment:
I tested this locally, apparently the way we had the migration initially did not actually restrict access to `open_graph_raw` and `open_graph_patch`, you can try this by going to dashboard.estuary.dev/connectors, and re-sending the request with an additional `open_graph_raw`, and you will get that column back.

In contrast, if you grant only specific columns and not the ones you want accessible by users, it seems to work. So trying to fetch the restricted columns I now get a 403.

cc @jgraettinger @psFried @saterus